src/contest_link_80F57C4.c

```diff
@@ -1607,9 +1591,10 @@ static void sub_80F75A8(struct Sprite *sprite)
     }
     else
     {
-        s16 delta = (u16)sprite->data[1] + 0x600;
+        s16 delta = sprite->data[1] + 0x600;
         sprite->pos1.x -= delta >> 8;
-        sprite->data[1] = (sprite->data[1] + 0x600) & 0xFF;
+        sprite->data[1] += 0x600;
+        sprite->data[1] &= 0xFF;
```

Comment:
I question GameFreak sometimes

Reply:
Me too, but thanks to them I can feel proud having figured out this specific compiler pattern